### PR TITLE
Add annotation to PodSpec on Deployment whenever ConfigMap changes

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Add annotation to PodSpec on Deployment whenever ConfigMap changes ResourceVersion changes to ensure Pod is redeployed

### CODE REVIEW

File: pkg/operator/deployment.go [hunk 1]
```diff
@@ -172,6 +180,26 @@ func consoleContainer(cr *v1alpha1.Console) corev1.Container {
 
 }
 
+func UpdateDeployment(cr *v1alpha1.Console, cm *corev1.ConfigMap) (*appsv1.Deployment, error) {
+	d := newConsoleDeployment(cr)
+	if err := sdk.Get(d); err != nil {
+		return nil, err
+	}
+
+	if len(d.Spec.Template.ObjectMeta.Annotations) == 0 {
```
Comment (enj) [inline pkg/operator/deployment.go]: this should be nil check

File: pkg/operator/deployment.go [hunk 2]
```diff
@@ -172,6 +180,26 @@ func consoleContainer(cr *v1alpha1.Console) corev1.Container {
...
+		d.Spec.Template.ObjectMeta.Annotations = map[string]string{}
+	}
+
+	// If the configmap is updated, we want to rollout a new pod.
+	// setting an annotation is sufficient for this task.
+	d.Spec.Template.ObjectMeta.Annotations[configMapResourceVersionAnnotation] = cm.ObjectMeta.GetResourceVersion()
+
+	if err := sdk.Update(d); err != nil {
```
Comment (enj) [inline pkg/operator/deployment.go]: Only update if changed

File: pkg/operator/deployment.go (hunk 1 above)
Comment (benjaminapetersen) [reply pkg/operator/deployment.go]: will do.

File: pkg/operator/deployment.go (hunk 2 above)
Comment (benjaminapetersen) [reply pkg/operator/deployment.go]: can do.

File: pkg/operator/deployment.go [hunk 3]
```diff
@@ -172,6 +180,35 @@ func consoleContainer(cr *v1alpha1.Console) corev1.Container {
 
 }
 
+func UpdateDeployment(cr *v1alpha1.Console, cm *corev1.ConfigMap) (*appsv1.Deployment, error) {
+	d := newConsoleDeployment(cr)
+	changed := false
```
Comment (benjaminapetersen) [inline pkg/operator/deployment.go]: @enj tracking changed here.

**Pull request merged**